pkg/provider/kubernetes/ingress/client_mock_test.go

```diff
@@ -105,6 +119,13 @@ func (c clientMock) GetSecret(namespace, name string) (*corev1.Secret, bool, err
 	return nil, false, nil
 }
 
+func (c clientMock) GetIngressClass(name string) (*networkingv1beta1.IngressClass, bool, error) {
```

Comment:
couldn't we instead / or additionally look for `IngressClass` Object with the controller `traefik.io/ingress-controller` in the spec?